src/words/environment/Environment.java

```diff
@@ -135,6 +135,24 @@ public WordsObject createObject(String objectName, String className, Position po
 		}
 	}
 	
+	public void removeObject(WordsObject object) {
+		object.clearReferers();
+		
+		for (HashMap<String, WordsObject> scope : objectsByName) {
+			if (scope.get(object.getObjectName()) != null) {
```

Comment:
I think technically this check is not necessary, since the remove() method says it removes a key if it exists and does nothing if not.